api/controllers/loomies.go

```diff
@@ -182,3 +182,71 @@ func HandleValidateLoomieExists(c *gin.Context) {
 		"loomie_id": loomie_id,
 	})
 }
+
+func HandleCaptureLoomie(c *gin.Context) {
+	loomie_id := interfaces.LoomieForm{}
+	userid, _ := c.Get("userid")
+
+	if err := c.BindJSON(&loomie_id); err != nil {
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": true, "message": "Bad request"})
+		return
+	}
+
+	if loomie_id.LoomieId == "" {
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": true, "message": "No Loomie ID was provided"})
+		return
+	}
+
+	loomie, err := models.ValidateLoomieExists(loomie_id.LoomieId)
+
+	if err != nil {
+		if err == mongo.ErrNoDocuments {
+			c.AbortWithStatusJSON(http.StatusNotFound, gin.H{"error": true, "message": "Loomie was not found"})
+			return
+		} else {
+			c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{"error": true, "message": "Internal server error"})
+			return
+		}
+	}
+
+	user, err := models.GetUserById(userid.(string))
+
+	if err != nil {
+		if err == mongo.ErrNoDocuments {
+			c.AbortWithStatusJSON(http.StatusNotFound, gin.H{"error": true, "message": "User was not found"})
+			return
+		} else {
+			c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{"error": true, "message": "Internal server error"})
+			return
+		}
+	}
+
+	caught_loomies := interfaces.CaughtLoomie{Owner: user.Id,
```

Comment:
You must also check if the user is near the loomie.

Reply:
How close should the user be to the loomie?

Reply:
There is already an environment variable to validate It. The max distancie is the "radius" of the zones. Check the endpoint to claim the gyms rewards, you'll find an example there.